mempool/mempool.go

```diff
@@ -105,7 +105,9 @@ type TxInfo struct {
 // PreCheckMaxBytes checks that the size of the transaction is smaller or equal to the expected maxBytes.
 func PreCheckMaxBytes(maxBytes int64) PreCheckFunc {
 	return func(tx types.Tx) error {
-		txSize := int64(len(tx))
+		txLength := types.ComputeProtoOverhead(tx, 1)
```

Comment:
I wonder if we can just use https://github.com/golang/protobuf/blob/91c84e0db17890c2bb64280a6d660e73e4237fd1/proto/wire.go#L13 `Size` function or other func exposed by golang protobuf. Have you thought about it?

Reply:
Yes, looked for a bit but to do this we would have to wrap the tx in proto.Message then call size on it. This way we avoid doing this

Reply:
But if the protobuf spec changes, we won't have to change our code

Reply:
Yeah, I'm inclined to agree with @melekes here.

Reply:
made the change, running tests locally. I agree as well